src/compiler/checker.ts

```diff
@@ -13175,6 +13177,21 @@ namespace ts {
             if (node.flags & NodeFlags.Abstract && node.body) {
                 error(node, Diagnostics.Method_0_cannot_have_an_implementation_because_it_is_marked_abstract, declarationNameToString(node.name));
             }
+
+            checkNameOfStaticMethodOrPropertyInClass(node);
+        }
+
+        function checkNameOfStaticMethodOrPropertyInClass(node: ClassElement) {

```

Comment:
We should do this in a more generic fashion and not hard code the names. we should use the `Function` type and make sure we are not redefining any of the properties there. possibly `readonly` ones only. this way users can add/remove some of these checks.